dotnet/src/Connectors/Connectors.Memory.Sqlite/Database.cs

```diff
@@ -73,6 +73,21 @@ public async Task CreateCollectionAsync(SqliteConnection conn, string collection
         await cmd.ExecuteNonQueryAsync(cancellationToken).ConfigureAwait(false);
     }
 
+    public async Task UpsertAsync(SqliteConnection conn,
+        string collection, string key, string? metadata, string? embedding, string? timestamp, CancellationToken cancellationToken = default)
+    {
+        using SqliteCommand cmd = conn.CreateCommand();
+        cmd.CommandText = $@"
+        INSERT OR REPLACE INTO {TableName}(collection, key, metadata, embedding, timestamp)
+        VALUES(@collection, @key, @metadata, @embedding, @timestamp);";
+        cmd.Parameters.AddWithValue("@collection", collection);
+        cmd.Parameters.AddWithValue("@key", key);
+        cmd.Parameters.AddWithValue("@metadata", metadata ?? string.Empty);
+        cmd.Parameters.AddWithValue("@embedding", embedding ?? string.Empty);
+        cmd.Parameters.AddWithValue("@timestamp", timestamp ?? string.Empty);
+        await cmd.ExecuteNonQueryAsync(cancellationToken).ConfigureAwait(false);
+    }
+
     public async Task InsertOrIgnoreAsync(SqliteConnection conn,
```

Comment:
@atiq-bs23 as you suggested, yes, please remove this method now that it's not required anymore.  Looks like UpdateAsync can be removed as well actually.

Also, thanks for contributing this improvement!

Reply:
Okay, removing

Reply:
Removed @westey-m